ChronoAPI/ChronoLog/include/ConfigurationManager.h

```diff
@@ -9,58 +9,110 @@
 #include <cstdlib>
 #include <fstream>
 #include <iostream>
-//#include <filesystem>
+#include <rapidjson/document.h>
+#include <sstream>
 #include "data_structures.h"
 #include "enum.h"
 #include "log.h"
+#include "errcode.h"
+#include "ClocksourceManager.h"
 
```

Comment:
ClocksourceManager should be part of ConfigurationManager.h
Configuration Manager is only responsible for reading in the configuration values from the JSON formatted file and doing basic validation on them ( valid ip string , etc)
ConfigurationManager should not be instantiating ClocksourceManager.  ConfigurationManager is a Singleton that all other modules will discover through it's GetInstance() function and pull the configuration values they need  from ConfigurationManager.